Change package name to match soname

mckees opened the pull request:
Hector pointed out that this was mismatched, and we originally underestimated how standard this is. With level zero being libze, I think this name is also pretty clear

mckees committed:
Change package name to match soname

mckees commented:
@androniychuk

androniychuk commented on debian/control:
```diff
@@ -16,7 +16,7 @@ Build-Depends: debhelper (>= 11),
 Homepage: https://github.com/intel/level-zero-raytracing-support
 
-Package: intel-level-zero-gpu-raytracing
+Package: libze-intel-gpu-raytracing
```
can we keep it as previous?

mckees replied:
We're explicitly overriding a Lintian warning without any real justification. I'd be surprised if that doesn't get pushback. It's the name that Intel has gone with for the soname, so I guess we could send a PR to change that soname in the source. Otherwise, we'd be diverging from the source

androniychuk replied:
ok, lets change, it
but we need to add a replaces and conflicts line with previous pkg, otherwise all users of PPA will get borked

mckees replied:
Added a "replaces" line and changed the base to oracular (Not sure why we even had a main branch)

androniychuk replied:
I noticed if there is no conflicts line, then apt repos don't handle it as smoothly as they could

mckees replied:
What is the behavior you're seeing?

androniychuk replied:
apt/dpkg install errors

mckees replied:
Ah, I see it now, thanks. At least on my end, that was a scarily silent since both got installed just fine. I'm going to ask around why "Replaces" on its own wouldn't remove the old one it replaces and install the new one. In the meantime, agreed that Conflicts is the best option. Pushed it up

mckees committed:
Add conflict with previous name

androniychuk commented on debian/control:
```diff
@@ -16,7 +16,9 @@ Build-Depends: debhelper (>= 11),
 
-Package: intel-level-zero-gpu-raytracing
+Package: libze-intel-gpu-raytracing
+Replaces: intel-level-zero-gpu-raytracing (>= 1.0.0-0ubuntu1~24.10~ppa4)
```
I wouldn't version this, just keep it unversioned

mckees replied:
Alright, changed that in the latest commit

mckees committed:
Remove versioning from Replaces

It is not necessary since all versions of
intel-level-zero-gpu-raytracing will be a lower version than all
versions of libze-intel-gpu-raytracing
